### ISSUE TITLE
`Exception: Can't split` serializing to xml term ending with a percent-encoded character

### ISSUE THREAD

**Issue by lrowe** [RDFLib/rdflib#561]
I get an exception xml serializing a term like `http://example.com/term%20` which ends with a percent encoded character:

``` python
>>> g = rdflib.Graph()
>>> g.add((rdflib.URIRef('http://example.com/foo'), rdflib.URIRef('http://example.com/term%20'), rdflib.URIRef('http://example.com/bar')))
>>> g.serialize(format='xml')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/lrowe/.buildout/eggs/rdflib-4.2.2.dev0-py3.5.egg/rdflib/graph.py", line 942, in serialize
    serializer.serialize(stream, base=base, encoding=encoding, **args)
  File "/Users/lrowe/.buildout/eggs/rdflib-4.2.2.dev0-py3.5.egg/rdflib/plugins/serializers/rdfxml.py", line 66, in serialize
    bindings = list(self.__bindings())
  File "/Users/lrowe/.buildout/eggs/rdflib-4.2.2.dev0-py3.5.egg/rdflib/plugins/serializers/rdfxml.py", line 33, in __bindings
    prefix, namespace, name = nm.compute_qname(predicate)
  File "/Users/lrowe/.buildout/eggs/rdflib-4.2.2.dev0-py3.5.egg/rdflib/namespace.py", line 335, in compute_qname
    namespace, name = split_uri(uri)
  File "/Users/lrowe/.buildout/eggs/rdflib-4.2.2.dev0-py3.5.egg/rdflib/namespace.py", line 507, in split_uri
    raise Exception("Can't split '%s'" % uri)
Exception: Can't split 'http://example.com/term%20'
```

However serializing to n3 works:

``` python
>>> g.serialize(format='n3')
b'@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .\n@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .\n@prefix xml: <http://www.w3.org/XML/1998/namespace> .\n@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .\n\n<http://example.com/foo> <http://example.com/term%20> <http://example.com/bar> .\n\n'
```

As does xml serializing of an unbased path like `/term%20`:

``` python
>>> g = rdflib.Graph()
>>> g.add((rdflib.URIRef('/foo'), rdflib.URIRef('/term%20'), rdflib.URIRef('/bar')))
>>> g.serialize(format='xml')
b'<?xml version="1.0" encoding="UTF-8"?>\n<rdf:RDF\n   xmlns:ns1="/"\n   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"\n>\n  <rdf:Description rdf:about="/foo">\n    <ns1:term%20 rdf:resource="/bar"/>\n  </rdf:Description>\n</rdf:RDF>\n'
```

**Comment by Ruthicaf** [comment]
I get the same exception. Still. 3 years later.
It appears that the `Namespace.split_uri()` method does not handle the percent sign in url-encoded URIs.
The URI's appear to be valid, as they pass the `_is_valid_uri()` check.
This should be an easy fix by updating the `ALLOWED_NAME_CHARS` to include the percent sign. 
If you wanted to make it more complicated you could make it only allow the percent sign in the context of a valid URL escape code.

The `n3()` method appears to work as it recovers from the error and returns the full, non-prefixed URI. 
This is not suitable for my needs. Please fix this issue.

**Comment by gjhiggins** [comment]
Closing because this issue appears to have been resolved in a subsequent commit because (in 0.6.1) the example code no longer raises an Exception:

```py
def test_xml_serialize():
    import rdflib

    g = rdflib.Graph()
    g.add(
        (
            rdflib.URIRef("http://example.com/foo"),
            rdflib.URIRef("http://example.com/term%20"),
            rdflib.URIRef("http://example.com/bar"),
        )
    )

    assert g.serialize(format="xml") == (
        '<?xml version="1.0" encoding="utf-8"?>\n'
        "<rdf:RDF\n"
        '   xmlns:ns1="http://example.com/"\n'
        '   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"\n'
        ">\n"
        '  <rdf:Description rdf:about="http://example.com/foo">\n'
        '    <ns1:term%20 rdf:resource="http://example.com/bar"/>\n'
        "  </rdf:Description>\n"
        "</rdf:RDF>\n"
    )
```
